packages/blade/src/components/SideNav/SideNavItems/SideNavLink.tsx

```diff
@@ -0,0 +1,327 @@
+        '&::before': {
+          opacity: 0,
+          transition: notchOpacity,
+        },
+      },
+    },
+  };
+});
+
+const TooltipifyNavLink = ({
+  children,
+  tooltip,
+}: {
+  children: React.ReactElement;
+  tooltip: SideNavLinkProps['tooltip'];
+}): React.ReactElement => {
+  if (!tooltip) {
+    return children;
+  }
+
+  return (
+    <Tooltip {...tooltip} placement="top">
+      {children}
+    </Tooltip>
+  );
+};
+
+const NavLinkIconTitle = ({
+  icon: Icon,
+  title,
+  titleSuffix,
+  isL1Item,
+}: Pick<SideNavLinkProps, 'title' | 'icon' | 'titleSuffix'> & {
+  isL1Item: boolean;
+}): React.ReactElement => {
+  return (
+    <Box display="flex" flexDirection="row" gap="spacing.3">
+      {Icon ? (
+        <BaseBox display="flex" flexDirection="row" alignItems="center" justifyContent="center">
+          <Icon size="medium" color="currentColor" />
+        </BaseBox>
+      ) : null}
+      <BaseText
+        truncateAfterLines={1}
+        color="currentColor"
+        fontWeight="medium"
+        fontSize={100}
+        lineHeight={100}
+        as="p"
+        className={isL1Item ? HIDE_WHEN_COLLAPSED : ''}
+      >
+        {title}
+      </BaseText>
+      {titleSuffix ? <BaseBox>{titleSuffix}</BaseBox> : null}
+    </Box>
+  );
+};
+
+const L3Trigger = ({
+  title,
+  icon,
+  as,
+  href,
+  titleSuffix,
+  tooltip,
+}: Pick<
+  SideNavLinkProps,
+  'title' | 'icon' | 'as' | 'href' | 'titleSuffix' | 'tooltip'
+>): React.ReactElement => {
+  const { onExpandChange, isExpanded, collapsibleBodyId } = useCollapsible();
+
+  const toggleCollapse = (): void => onExpandChange(!isExpanded);
+  const iconProps = {
+    size: 'medium',
+    color: 'currentColor',
+  } as const;
+
+  return (
+    <TooltipifyNavLink tooltip={tooltip}>
+      <StyledNavLinkContainer
+        as={href ? as : 'button'}
+        to={href}
+        onClick={toggleCollapse}
+        {...makeAccessible({ expanded: isExpanded, controls: collapsibleBodyId })}
+      >
+        <NavLinkIconTitle title={title} icon={icon} isL1Item={false} titleSuffix={titleSuffix} />
+        <BaseBox display="flex" alignItems="center">
+          {isExpanded ? <ChevronUpIcon {...iconProps} /> : <ChevronDownIcon {...iconProps} />}
+        </BaseBox>
+      </StyledNavLinkContainer>
+    </TooltipifyNavLink>
+  );
+};
+
+const CurvedVerticalLine = styled(BaseBox)((props) => {
+  const { colors, border, spacing } = props.theme;
+  return {
+    borderWidth: makeBorderSize(props.theme.border.width.thin),
+    borderColor: `${colors.transparent} ${colors.transparent} ${colors.surface.border.primary.muted} ${colors.surface.border.primary.muted}`,
+    borderStyle: 'solid',
+    borderRadius: `${makeBorderSize(border.radius.none)} ${makeBorderSize(
+      border.radius.none,
+    )} ${makeBorderSize(border.radius.none)} ${makeBorderSize(border.radius.medium)}`,
+    height: '100vh',
+    position: 'absolute',
+    top: `calc(-100vh + ${makeSize(NAV_ITEM_HEIGHT / 2)})`,
```

Comment:
Oh interesting. Add a comment on why this is like this.

Reply:
thank you for finding this interesting. added comment